R/FindRecalSeries.R

```diff
@@ -0,0 +1,107 @@
+#' Attempts to find most suitable series for recalibration.
+#'
+#' This function takes on input the CH2 homologous recalibration series, which are provided by the RecalList function #' and tries to find the most suitable series combination for recalibration based on the following criteria:
+#' 1) Series should cover the full mass spectral range
+#' 2) Series should be optimally long and combined have a “Tall Peak” at least every 100 m/z.
+#' 3) Abundance score: the higher, the better
+#' 4) Peak score: the closer to 0, the better
+#' 5) Peak Distance: the closer to 1, the better
+#' 6) Series Score: the closer to this value, the better 
+#' 
+#' The recal function can take up to 10 series - due to the size of the search space when looking for combinations of 10
+#' elements, a pre-filtering is done: only the series which have Abundance score > 100 are considered and the one #'
+#' having Peak Distance < 2. 
+#' Combinations of 5 series are assembled, scores are computed for other metrics (in case of Peak proximity and Peak
+#' distance, an inverted score is computed) and these are summed. Finally, top 10 unique series having the highest
+#' score are outputted.
+#' 
+#' Warning: this step is in general computationally demanding, for ~30 series it took around 30 min. 
+#' 
+#' @param df An output from RecalList, containing recalibrant CH2 series.
+#' @return A dataframe of 10 best-scoring series.
+
+findSeries <- function(df) {
+
+# Arrange the data
+df <- df %>%
+  separate(col = Mass.Range, into = c('Min.Mass.Range', 'Max.Mass.Range'), sep = "-") %>%
+  mutate(Min.Mass.Range = as.numeric(Min.Mass.Range), 
+         Max.Mass.Range = as.numeric(Max.Mass.Range)) %>%
+  mutate(Series.Length = Max.Mass.Range - Min.Mass.Range) %>%
+  filter(Abundance.Score > 100) %>%
+  filter(Peak.Distance < 2) 
```

Comment:
This should be done first and the values should be parameters

Reply:
resolved by 52978bad5fc76a952bb4fe224fe59d1e0fcdc602